File: Backend/BackendAPI/Models/Poll.cs
namespace BackendAPI.Models
{
    public class Poll
    {
        public long Id { get; set; }
        public string Question { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public string Category { get; set; } = string.Empty;
        public bool IsTrending { get; set; }
        public bool IsActive { get; set; } = true;
        public DateTime ExpiresAt { get; set; }
        public DateTime CreatedAt { get; set; }
        public int TotalVotes { get; set; }
        public List<PollOption> Options { get; set; } = new();
    }

    public class PollOption
    {
        public long Id { get; set; }
        public long PollId { get; set; }
        public string Text { get; set; } = string.Empty;
        public int VoteCount { get; set; }
        public double VotePercentage { get; set; }
    }

    public class CreatePollRequest
    {
        public string Question { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public string Category { get; set; } = string.Empty;
        public DateTime ExpiresAt { get; set; }
        public List<string> Options { get; set; } = new();
    }
}

File: apps/mobile/src/types/poll.ts
export interface ApiPollOption {
  id: number;
  pollId: number;
  text: string;
  voteCount: number;
  votePercentage: number;
}

export interface ApiPoll {
  id: number;
  question: string;
  description: string;
  category: string;
  isTrending: boolean;
  isActive: boolean;
  expiresAt: string;
  createdAt: string;
  totalVotes: number;
  createdByUserId: number | null;
  createdByUsername: string | null;
  createdByDisplayName: string | null;
  sourceType: string | null;
  sourceUrl: string | null;
  thumbnailUrl: string | null;
  isAIGenerated: boolean;
  options: ApiPollOption[];
  hasVoted: boolean;
  userVotedOptionId: number | null;
}

export interface VoteReward {
  xp: number;
  streak: number;
  longestStreak: number;
  totalVotes: number;
  xpAwarded: number;
  streakAdvanced: boolean;
  todayComplete: boolean;
  recoveryEligible: boolean;
  recoveryUsed: boolean;
  nextRecoveryAt: string | null;
  milestoneReached: number | null;
  lastVoteDate: string | null;
}

export interface CastVoteResponse {
  poll: ApiPoll;
  reward: VoteReward;
}
